low-level/files/opt/seagate/sspl/setup/sspl_cleanup.py

```diff
@@ -0,0 +1,123 @@
+#!/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import shutil
+import os
+import pwd
+
+from cortx.utils.conf_store import Conf
+from cortx.utils.process import SimpleProcess
+from cortx.utils.message_bus import MessageBus, MessageBusAdmin
+from cortx.utils.message_bus.error import MessageBusError
+from framework.utils.file_utils import FileUtils
+from files.opt.seagate.sspl.setup.setup_logger import logger
+from framework.utils.utility import Utility
+from framework.base.sspl_constants import (
+    SSPL_CONFIG_INDEX, file_store_config_path, global_config_file_path,
+    sspl_config_path, PRODUCT_BASE_DIR, SSPL_BASE_DIR, PRODUCT_FAMILY,
+    RSYSLOG_IEM_CONF, RSYSLOG_SSPL_CONF, RSYSLOG_MSB_CONF,
+    IEM_LOGROTATE_CONF, SSPL_LOGROTATE_CONF, MSB_LOGROTATE_CONF,
+    RSYSLOG_SB_CONF, SB_LOGROTATE_CONF, USER)
+
+
+class SSPLCleanup:
+
+    """Reset config and optionally factory operation."""
+
+    def __init__(self, args):
+        """Check '--pre-factory' in args."""
+        self.pre_factory = False
+        if "--pre-factory" in args:
+            self.pre_factory = True
+
+    def validate(self):
```

Comment:
Add generic docstring to this method even we don't have at this time.

Reply:
done.

Reply:
done.